src/blob.rs

```diff
@@ -452,7 +452,7 @@ impl<'a> BlobObject<'a> {
             img.write_to(encoded, image::ImageFormat::Jpeg)?;
             Ok(())
         }
-        fn encode_img_exceeds_bytes(
+        fn encoded_img_exceeds_bytes(
```

Comment:
not sure if the new name is better. the function still _encodes_ an image and does not only check if an _encoded image_ exceeds some bytes as the new name implies imho.

maybe sth. as `encode_image_and_check_byte_size` would describe the function better. 

just some thoughts. otoh, it is only an inner function used as a shortcut - so, at the end, i would be fine with all variations, this is not a blocker in any way :)

Reply:
The reason for changing was: I tried to make it clear that the fn does not _always_ encode the image, because that's the wrong assumption that caused the bug in the first place. It just encodes it if necessary, and in this case it writes the result into `encoded` for performance reasons (so that it doesn't have to be encoded again later). 

OTOH, you are right that the fn still _sometimes_ does encode the image.

Reply:
i see :)